File: ProyectoIntegradoFlavorSync/src/main/java/com/example/demo/entity/nutritionalInformation.java
package com.example.demo.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Positive;
import lombok.Data;

@Entity
@Table(name = "nutritionalInformation")
@Data
public class nutritionalInformation {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	// Calorías por porción
	@Column(name = "calories")
	@Positive(message = "Calories must be a positive number")
	private Float calories;

	// Grasas en gramos
	@Column(name = "fat")
	@Positive(message = "Fat content must be a positive number")
	private Float fat;

	// Carbohidratos en gramos
	@Column(name = "carbohydrates")
	@Positive(message = "Carbohydrates content must be a positive number")
	private Float carbohydrates;

	// Proteínas en gramos
	@Column(name = "proteins")
	@Positive(message = "Proteins content must be a positive number")
	private Float proteins;

	// Otros nutrientes (opcional, por ejemplo, fibra, sodio, etc.)
	@Column(name = "otherNutrients")
	private String otherNutrients;
}

File: Flavorsync/flavorsync/lib/models/nutritionalInformation.dart
import 'dart:convert';

class NutritionalInformation {
  // Identificador único para la información nutricional.
  int? id;

  // Calorías por porción
  double calories;

  // Grasas en gramos
  double fat;

  // Carbohidratos en gramos
  double carbohydrates;

  // Proteínas en gramos
  double proteins;

  // Otros nutrientes (opcional, por ejemplo, fibra, sodio, etc.)
  String? otherNutrients;

  // Constructor
  NutritionalInformation({
    this.id,
    required this.calories,
    required this.fat,
    required this.carbohydrates,
    required this.proteins,
    this.otherNutrients,
  });

  // Factory constructor to create an instance from a map (for JSON parsing)
  factory NutritionalInformation.fromMap(Map<String, dynamic> map) {
    return NutritionalInformation(
      id: map['id'],
      calories: map['calories']?.toDouble() ?? 0.0,
      fat: map['fat']?.toDouble() ?? 0.0,
      carbohydrates: map['carbohydrates']?.toDouble() ?? 0.0,
      proteins: map['proteins']?.toDouble() ?? 0.0,
      otherNutrients: map['otherNutrients'],
    );
  }

  // Method to convert the instance to a map (for JSON serialization)
  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'calories': calories,
      'fat': fat,
      'carbohydrates': carbohydrates,
      'proteins': proteins,
      'otherNutrients': otherNutrients,
    };
  }

  // Converts the object to a JSON string
  String toJson() => jsonEncode(toMap());

  // Creates an instance from a JSON string
  factory NutritionalInformation.fromJson(String source) => NutritionalInformation.fromMap(jsonDecode(source));

  @override
  String toString() {
    return 'NutritionalInformation(id: $id, calories: $calories, fat: $fat, carbohydrates: $carbohydrates, proteins: $proteins)';
  }
}
